src/main/java/ortus/boxlang/runtime/types/Query.java

```diff
@@ -779,4 +780,21 @@ public IStruct getMetaData() {
 		return meta;
 	}
 
+	public Query duplicate() {
+		return duplicate( false );
+	}
+
+	public Query duplicate( boolean deep ) {
+		Query q = new Query();
+		for ( var entry : this.getColumns().entrySet() ) {
```

Comment:
Can you check if the size of the query is > 50 rows and if so, use a parallel stream

Reply:
So this is just adding columns.  So you want that if the query is more than 50 columns?